rfcs/45-stageless.md

```diff
@@ -0,0 +1,823 @@
+      (check_if_projectiles_hit, despawn_projectiles_that_hit)
+      .chain()
+      .after_and_flush(spawn_projectiles).after(CommandFlush::PostUpdate)
+    )
+    // If we need to add more command flushes to make our logic work, we can manually insert them
+    .add_system(flush_commands.label("DespawningFlush").after(CommandFlush::PostUpdate).before(CommandFlush::EndOfFrame))
+    .add_system(fork_projectiles.after("DespawningFlush"));
+  }
+}
+```
+
+### Run criteria
+
+While ordering constraints determine *when* a system will run, **run criteria** will determine *if* it will run at all.
+Each system can have any number of run criteria, which read data from the world to control whether or not that system runs.
+If (and only if) all of its run criteria return `true`, the system will run.
+If any of the run criteria are `false`, the system will be skipped.
+Systems that are skipped are considered completed for the purposes of ordering constraints.
+
+Run criteria are not systems, but can be created from systems which can read (but not write) data from the `World` and returns a boolean value.
+
+You can specify run criteria in several different ways:
+
+```rust
+// This is just an ordinary system: timers need to be ticked!
+fn tick_construction_timer(timer: ResMut<ConstructionTimer>, time: Res<Time>){
+    timer.tick(time.delta());
+}
+
+// This function can be used as a run criterion system,
+// because it does not mutate data and returns `bool`
+fn construction_timer_finished(timer: Res<ConstructionTimer>) -> bool {
+    timer.finished()
+}
+
+// You can use queries, event readers and resources with arbitrarily complex internal logic
+fn too_many_enemies(population_query: Query<(), With<Enemy>>, population_limit: Res<PopulationLimit>) -> bool {
+   population_query.iter().count() > population_limit
+}
+
+fn main(){
+    App::new()
+    .add_plugins(DefaultPlugins)
+    // We can add functions with read-only system parameters as run criteria
+    .add_system((
+       tick_construction_timer, 
+       update_construction_progress)
+       .chain()
+       .run_if(construction_timer_finished)
+    )
+   .add_system(system_meltdown_mitigation.run_if(too_many_enemies))
+    // We can use closures for simple one-off run criteria, 
+    // which automatically fetch the appropriate data from the `World`
+    .add_system(spawn_more_enemies.run_if(|difficulty: Res<Difficulty>| difficulty >= 9000))
+    // The `run_if_resource_equals` method is fast, special-cased syntax that generates a run criterion
+    // for when you want to check the value of a resource (commonly an enum)
+    .add_system(gravity.run_if_resource_equals(Gravity::Enabled))
+    // Run criteria can be attached to labels: a copy of the run criteria will be applied to each system with that label
+    .add_label(GameLabel::Physics.run_if_resource_equals(Paused(false)))
+    .run();
+}
+```
+
+There are a few important subtleties to bear in mind when working with run criteria:

```

Comment:
I think it would be worth mentioning the implications for run criteria when using potentially blink-and-you'll-miss-it system parameters/filters like `Added<>`, `Changed<>`, or event readers.

Ex, if the system I'm applying run criteria to receives events, will it miss the events that trigger on the updates that the run criteria is false?

Reply:
I see. I think it would be worth mentioning this behavior here since this is the user-perspective portion of the RFC